### DISCUSSION TITLE
Lesson 7: Unable to deploy the VRFCoordinatorV2Mock

### DISCUSSION THREAD

**Discussion by kailashchoudhary11** [smartcontractkit/full-blockchain-solidity-course-py#1831] [Q&A] [answered]
I am trying to deploy this mock from here 

https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/mocks/VRFCoordinatorV2Mock.sol
But I am getting this error
 
RFCoordinatorV2Mock.constructor confirmed (code size to deposit exceeds maximum code size)   Block: 1   Gas used: 1000000 (100.00%)

I also tried to reduce the size of code by 50 lines by removing the implementations of functions which I did not require but still getting the same error.

May anyone help to resolve this issue?

**Comment by cromewar** [comment, marked as answer]
just a question, on which network are you deploying to?

**Reply by kailashchoudhary11** [reply]
@cromewar I am trying to deploy on the development network

**Reply by cromewar** [reply]
Try deleting the build folder on your project, I'm trying to reproduce this problem but I'm having hard time. Did you clone the mentioned repo and try to compile it?

**Reply by kailashchoudhary11** [reply]
Yes I am able to compile and clone the mentioned repo Also able to deploy the mocks. So there is definitely something wrong with my code I think.

**Reply by kailashchoudhary11** [reply]
My code is getting deployed if I replace the code inside the addConsumer Function of VRFCoordinatorV2Mock with `revert("not implemented");` this line. But then I am not able to add consumer 😬 I feel so frustrated now I have been trying to make it work from so many days 🥲

**Reply by cromewar** [reply]
You can just copy the mock code on your code and try to run it

**Reply by kailashchoudhary11** [reply]
That did not work as well. Now I created a virtual environment and then installed brownie in that environment and created a new brownie project. I am able to deploy in this new project.

**Reply by cromewar** [reply]
jum, so it was a problem with your brownie env at the end. Sorry for not being able to provide more help on that one as the problems seems not to be code related but more environment related. I will set this as answered, please feel free to ask any other question on the repo.
